config/schemas/operator.schema

```diff
@@ -0,0 +1,219 @@
+          "$ref": "#/definitions/TaskSpec"
+        }
+      },
+      "type": "object",
+      "description": "a collection of Tasks to be used in the Steps of Plans mapped by a name"
+    },
+    "url": {
+      "type": "string",
+      "description": "project URL for the operator",
+      "examples": [
+        "http://kudo.dev"
+      ]
+    },
+    "version": {
+      "type": "string",
+      "description": "version of the operator must be semver x.y.z",
+      "examples": [
+        "0.1.1"
+      ]
+    }
+  },
+  "additionalProperties": false,
+  "type": "object",
+  "definitions": {
+    "Maintainer": {
+      "properties": {
+        "email": {
+          "type": "string",
+          "description": "email for maintainer",
+          "examples": [
+            "billy@kudo.dev"
+          ]
+        },
+        "name": {
+          "type": "string",
+          "description": "name of maintainer",
+          "examples": [
+            "Billy Bob"
+          ]
+        }
+      },
+      "additionalProperties": false,
+      "type": "object"
+    },
+    "OperatorDependency": {
+      "additionalProperties": true,
+      "type": "object"
+    },
+    "Phase": {
+      "required": [
+        "name",
+        "strategy",
+        "steps"
+      ],
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "name of the phase",
+          "examples": [
+            "deploy-zookeeper"
+          ]
+        },
+        "steps": {
+          "items": {
+            "$schema": "http://json-schema.org/draft-04/schema#",
+            "$ref": "#/definitions/Step"
+          },
+          "type": "array",
+          "description": "list of steps for this phase"
+        },
+        "strategy": {
+          "type": "string",
+          "description": "ordering strategy for steps in this phase",
+          "examples": [
+            "serial",
+            "parallel"
+          ]
+        }
+      },
+      "additionalProperties": false,
+      "type": "object"
+    },
+    "Plan": {
+      "required": [
+        "strategy",
+        "phases"
+      ],
+      "properties": {
+        "phases": {
+          "items": {
+            "$schema": "http://json-schema.org/draft-04/schema#",
+            "$ref": "#/definitions/Phase"
+          },
+          "type": "array",
+          "description": "list of phases for this plan"
+        },
+        "strategy": {
+          "type": "string",
+          "description": "ordering strategy for phases in this plan",
+          "examples": [
+            "serial",
+            "parallel"
+          ]
+        }
+      },
+      "additionalProperties": false,
+      "type": "object"
+    },
+    "Step": {
+      "required": [
+        "name",
+        "tasks"
+      ],
+      "properties": {
+        "delete": {
+          "type": "boolean",
+          "description": "step requires a kubernetes delete instead of apply"
```

Comment:
still not clear what the `delete` portion will do here when I read the description, can we phrase it different?